nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java

```diff
@@ -230,78 +246,97 @@ private boolean optionalIsGetCall(Symbol.MethodSymbol symbol, Types types) {
    * A {@link VariableElement} for a dummy "field" holding the contents of an Optional object, used
    * in dataflow analysis to track whether the Optional content is present.
    */
-  private static VariableElement getOptionalContentElement() {
-    return new VariableElement() {
-      @Override
-      @Nullable
-      public Object getConstantValue() {
-        return null;
+  private static final class OptionalContentVariableElement implements VariableElement {
+    public static final Context.Key<OptionalContentVariableElement> contextKey =
+        new Context.Key<>();
+
+    private static final Set<Modifier> MODIFIERS = ImmutableSet.of(Modifier.PUBLIC, Modifier.FINAL);
+    private final Name name;
+    private final TypeMirror asType;
+
+    static synchronized VariableElement instance(Context context) {
+      OptionalContentVariableElement instance = context.get(contextKey);
+      if (instance == null) {
+        instance =
+            new OptionalContentVariableElement(
+                Names.instance(context).fromString("value"), Symtab.instance(context).objectType);
+        context.put(contextKey, instance);
       }
+      return instance;
+    }
 
-      @Override
-      @Nullable
-      public Name getSimpleName() {
-        return null;
-      }
+    private OptionalContentVariableElement(Name name, TypeMirror asType) {
+      this.name = name;
+      this.asType = asType;
+    }
 
-      @Override
-      @Nullable
-      public Element getEnclosingElement() {
-        return null;
-      }
+    @Override
+    @Nullable
+    public Object getConstantValue() {
+      return null;
+    }
 
-      @Override
-      @Nullable
-      public List<? extends Element> getEnclosedElements() {
-        return null;
-      }
+    @Override
+    public Name getSimpleName() {
+      return name;
+    }
 
-      @Override
-      @Nullable
-      public List<? extends AnnotationMirror> getAnnotationMirrors() {
-        return null;
-      }
+    @Override
+    @Nullable
+    public Element getEnclosingElement() {
+      // A field would have an enclosing element, however this method isn't guaranteed to
+      // return non-null in all cases. It may be beneficial to implement this in a future
+      // improvement.
+      return null;
```

Comment:
This could be tricky to implement actually, since we support multiple `Optional` types (Java's and Guava's, e.g.).  If you agree, maybe we could mention that as the reason we return `null`?